data/org.freedesktop.portal.InputCapture.xml

```diff
@@ -0,0 +1,547 @@
+
+        Pointer barriers are situated on the top or left of their respective pixels and a
+        pointer barrier must be within or at the extent of the available
+        zone. For example, a zone of 1920x1080 at offset 0,0 allows
+        a pointer barrier at x1=1920, y1=0, x2=1920, y2=1080.
+        A pointer barrier at the extent of a zone is considered triggered
+        when the pointer would logically move off that zone, even if the
+        actual cusor movement is clipped to the zone.
+
+        A zero-sized array of pointer barriers removes all existing pointer barriers
+        for this session. Setting pointer barriers immediately suspends the
+        current session and the application must call Enable() after this method.
+
+        The @serial must be equivalent to the last returned serial number of the
+        org.freedesktop.portal.CpatureInput.GetZones() method. A serial mismatch
+        implies the application is not using the current zone set and
+        pointer barriers will fail.
+
+        Supported keys in the @options vardict include:
+        <variablelist>
+          <varlistentry>
+            <term>handle_token s</term>
+            <listitem><para>
+              A string that will be used as the last element of the @handle. Must be a valid
+              object path element. See the #org.freedesktop.portal.Request documentation for
+              more information about the @handle.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+
+        Supported keys in the @barriers vardicts include:
+        <variablelist>
+          <varlistentry>
+            <term>barrier-id u</term>
+            <listitem><para>
+              The non-zero id of this barrier. This id is used in the
+              #org.freedesktop.portal.InputCapture::Activated signal to inform
+              which barrier triggered input capture.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>position iiii</term>
+            <listitem><para>
+              The x1/y1 x2/y2 position of the pointer barrier. A horizontal
+              pointer barrier must have y1 == y2, a vertical pointer barrier
+              must have x1 == x2. Diagonal pointer barriers are not supported.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>direction i</term>
+            <listitem><para>
+                The directional movement to trigger the pointer barrier. A
+                negative integer implies the pointer barrier triggers on negative
+                movement across the pointer barrier (e.g. "up"
+                for a horizontal barrier), a positive integer implies the
+                pointer barrier triggers on positive movement across
+                the pointer barrier (e.g. "right" for a vertical barrier).
+                A direction of zero triggers the barrier in either direction.
```

Comment:
Should we make this a "enum" i.e. a `u` where we define `positive` and `negative`? Alt. limit it to `-1` and `1` so that `-42` and `123` aren't valid directions.

Reply:
good point, will amend

Reply:
On the other hand, with barriers only being allowed on the outer edges of all the zones, the direction isn't needed as only one direction makes any sense.

Reply:
done!